Sources/Swift/Tools/SentryLogBatcher.swift

```diff
@@ -6,24 +6,120 @@ import Foundation
 @_spi(Private) public class SentryLogBatcher: NSObject {
     
     private let client: SentryClient
+    private let flushTimeout: TimeInterval
+    private let maxBufferSizeBytes: Int
+    private let dispatchQueue: SentryDispatchQueueWrapper
     
-    @_spi(Private) public init(client: SentryClient) {
+    // All mutable state is accessed from the same dispatch queue.
+    private var encodedLogs: [Data] = []
+    private var encodedLogsSize: Int = 0
+    private var timerWorkItem: DispatchWorkItem?
+
+    @_spi(Private) public init(
+        client: SentryClient,
+        flushTimeout: TimeInterval,
+        maxBufferSizeBytes: Int,
+        dispatchQueue: SentryDispatchQueueWrapper
+    ) {
         self.client = client
+        self.flushTimeout = flushTimeout
+        self.maxBufferSizeBytes = maxBufferSizeBytes
+        self.dispatchQueue = dispatchQueue
         super.init()
     }
     
+    @_spi(Private) public convenience init(client: SentryClient, dispatchQueue: SentryDispatchQueueWrapper) {
+        self.init(
+            client: client,
+            flushTimeout: 5,
+            maxBufferSizeBytes: 1_024 * 1_024, // 1MB
+            dispatchQueue: dispatchQueue
+        )
+    }
+    
     func add(_ log: SentryLog) {
-        dispatch(logs: [log])
+        dispatchQueue.dispatchAsync { [weak self] in
+            self?.encodeAndBuffer(log: log)
+        }
     }
     
-    private func dispatch(logs: [SentryLog]) {
+    @objc
+    public func flush() {
```

Comment:
`l`: This is an async flush. If we want to offer a blocking flush that you also call from SentrySDK.flush we need to make this sync. But we should follow up and it's not required for the first experimental release.

Reply:
Added followup issue #5661